viam-cartographer/src/carto_facade/carto_facade.cc

```diff
@@ -84,9 +102,7 @@ config from_viam_carto_config(viam_carto_config vcc) {
     for (int i = 0; i < vcc.sensors_len; i++) {
         c.sensors.push_back(to_std_string(vcc.sensors[i]));
     }
-
     c.data_dir = to_std_string(vcc.data_dir);
-    c.component_reference = to_std_string(vcc.component_reference);
```

Comment:
turns out that having a component_reference doesn't make sense given it is just always the first element in the sensors list